### PR TITLE
Next try remove parameter

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
So lets try it again.

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
No we have got a the same result based on a green master ... :-(

**Comment by ElectricMaxxx** [comment]
@damienflament you gave a help with some tests on `routing-auto` can you give some input what is wrong here?

I debugged the first test and after renaming the blog, the blog route is there but the post routes are really gone. I can't really find a point where the route is moved, it seams as there route was newly created.

**Comment by damienflament** [comment]
Not at home for now. In will look at it this evening.

**Comment by damienflament** [comment]
@ElectricMaxxx : I also start debugging the first test and came to the same conclusion (by checking the SQlite database while debugging): the route for the blog is updated but the route for the post is not.

As your changes are minor, I see only 2 possibilities:

1. The content document contained in the `UriContext` is not the same as the one referenced by the removed `$contentDocument` parameter in the `PhpcrOdmAdapter::createAutoRoute` method and it results in an unexpected behavior. As this method is called by the `AutoRouteManager`, the issue should be in the _routing-auto_ component,
2. The issue is not the one described above. But as the only major change in your PR is a component version bump to RC3, the issue should be somewhere in the component code.

Let's eat something then I will debug deeper in the component source code.

**P.S:** Do you find any clue ?

**Comment by ElectricMaxxx** [comment]
@damienflament sounds valid. I got not that experience on routing-auto to see through the complete mechanism. But The erros feels like you described. I went back step by step locally and only when the $contentDocument passed as a parameter was used, it worked again.

**Comment by damienflament** [comment]
@ElectricMaxxx

> I went back step by step locally and only when the $contentDocument passed as a parameter was used, it worked again.

What did you do to put back the `$contentDocument` parameter ? As it is defined in the `AdapterInterface`,
you should:
- modify the `createAutoRoute` method signature both in the interface and the class implementing it (`PhpcrOdmAdapter`),
- OR modify the signature in the `PhpcrOdmAdapter` and set the component required version back to RC2.

**Comment by ElectricMaxxx** [comment]
I tried both, changing interface in vendor and/or moving back to RC2. -------- Originalnachricht --------Betreff: Re: [symfony-cmf/routing-auto-bundle] Next try remove parameter (#211)Von: Damien Flament An: symfony-cmf/routing-auto-bundle Cc: Maximilian Berghoff ,Mention @ElectricMaxxx

I went back step by step locally and only when the $contentDocument passed as a parameter was used, it worked again.

What did you do to put back the $contentDocument parameter ? As it is defined in the AdapterInterface,
you should:

modify the createAutoRoute method signature both in the interface and the class implementing it (PhpcrOdmAdapter),
OR modify the signature in the PhpcrOdmAdapter and set the component required version back to RC2.


—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/symfony-cmf/routing-auto-bundle","title":"symfony-cmf/routing-auto-bundle","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/symfony-cmf/routing-auto-bundle"}},"updates":{"snippets":[{"icon":"PERSON","message":"@damienflament in #211: @ElectricMaxxx\r\n\r\n\u003e I went back step by step locally and only when the $contentDocument passed as a parameter was used, it worked again.\r\n\r\nWhat did you do to put back the `$contentDocument` parameter ? As it is defined in the `AdapterInterface`,\r\nyou should:\r\n- modify the `createAutoRoute` method signature both in the interface and the class implementing it (`PhpcrOdmAdapter`),\r\n- OR modify the signature in the `PhpcrOdmAdapter` and set the component required version back to RC2."}],"action":{"name":"View Pull Request","url":"https://github.com/symfony-cmf/routing-auto-bundle/pull/211#issuecomment-322311902"}}}

**Comment by damienflament** [comment]
Hmm... Strange... As shown by [this tag comparison](https://github.com/symfony-cmf/routing-auto/compare/2.0.0-RC2...2.0.0-RC3) on the component, the changes in RC3 should not modify the behaviour. They are most changes in the interfaces, tests, dependency version.

Only [one change](https://github.com/symfony-cmf/routing-auto/compare/2.0.0-RC2...2.0.0-RC3#diff-d629aeb5f2f1f0a289901752fd465026R160) can modify the behavior. When getting a autoroute by locale in a context collection, if the locale of the context is no the same as the one from the autoroute.

In your PR, as said in a previous comment, the only change is when [getting the content document from the context](https://github.com/symfony-cmf/routing-auto-bundle/pull/211#issuecomment-322291564).

**Comment by damienflament** [comment]
In the `UriContextCollection::getAutoRouteByLocale` method, comparing the searched locale with the one from the autoroute (instead of the one from the context containing the autoroute) [on this line](https://github.com/symfony-cmf/routing-auto/compare/2.0.0-RC2...2.0.0-RC3#diff-d629aeb5f2f1f0a289901752fd465026R160) makes the tests pass.

This can't be detected by the `UriContextCollectionTest` has it considers that the context locale matches the locale of the autoroute. But it seems this is not always true.

@dantleech, @dbu : Any idea about the reason of this difference ? Is this behaviour possible and expected ?

**Comment by damienflament** [comment]
OK. Found it. It's about the `no-multilang` locale value.
The autoroute locale is `no-multilang` but the context containing the autoroute has a `null` locale.

But why ?

**Comment by dbu** [comment]
the commit has been done by you ;-) you even commented on that line: https://github.com/symfony-cmf/routing-auto/pull/95#discussion_r122610704

i am not fully on top of what the different pieces (context, the manager and collection and collectionbuilder) all do, but it seems to me we should go back to ask the autoroute for its locale. possibly its always in sync in a real application and just a problem how we mock data.

**Comment by dbu** [comment]
created https://github.com/symfony-cmf/routing-auto/pull/102 and will now change what we test with here

**Comment by damienflament** [comment]
I know about the change I done. The real modification was to check for the autoroute contained in the context to not be `null`. I compare with the locale of the context because I considered it matching the locale of the autoroute.

**Comment by dbu** [comment]
the phpdoc on the context says "Return the locale of the translated subject. Is null if the subject should not be translated." so its maybe not about the route. but no idea if that is a real difference or if this must always end up being the same in a real application.

**Comment by damienflament** [comment]
Right! So it is not necessary related to the autoroute, but to the subject.

File: composer.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "require": {
         "php": "^5.6|^7.0",
         "symfony/framework-bundle": "^2.8|^3.0",
-        "symfony-cmf/routing-auto": "2.0.0-RC2",
+        "symfony-cmf/routing-auto": "dev-locale-of-autoroute as 2.0.0-RC3",
```
Comment (dbu) [inline composer.json]: revert before merging

**Comment by ElectricMaxxx** [comment]
@damienflament maybe i can invite you the `symfony_cmf` channel in symfony slack?

What do we have to change here now? Release on `routing-auto` and restart builds?

**Comment by ElectricMaxxx** [comment]
Schall we revert my revert of the jms dependency change i thought which was the first sulution?

**Comment by damienflament** [comment]
@ElectricMaxxx : Never used Slack. I tried Gitter recently. But it's a way to start using it!

**Comment by dbu** [comment]
re reverting the jms dependency: yep, trying that in https://github.com/symfony-cmf/routing-auto/pull/104

File: composer.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "require": {
         "php": "^5.6|^7.0",
         "symfony/framework-bundle": "^2.8|^3.0",
-        "symfony-cmf/routing-auto": "2.0.0-RC2",
+        "symfony-cmf/routing-auto": "dev-jms-metadata-version as 2.0.0-RC3",
```
Comment (dbu) [inline composer.json]: tag RC4 and change this line before we merge

Comment (ElectricMaxxx) [inline composer.json]: @dbu RC4? That is my last release, i think we should put your changes into a RC5?

Comment (dbu) [inline composer.json]: yes, it will be RC5

**Comment by ElectricMaxxx** [comment]
I increased the version, waiting for its release, but now i see the commits twice?

**Comment by dbu** [comment]
maybe because i rebased at some point and you did `git pull` instead of `git pull --rebase`? but we can squash the commits anyways

**Comment by ElectricMaxxx** [comment]
@dbu i did so. We do have 2 commits only now. I wait for RC5 on `routing-auto`.

**Comment by ElectricMaxxx** [comment]
Jippi.

**Pull request merged**